Login modal window

jotakar opened the pull request:
This request is for get a modal form window to login in  services like ftp, sftp, non OAuth.
For login to service like GitHub or Dropbox it uses a blank window.
It uses a file where the modal window is created using nested layers, whose styles are incorporated into styles.scss
I've tried to make the minimal modifications, for this reason the app keeps in using a window object as basis for the login process.

jotakar committed:
Cange Unifile for modal login

Modify to use a modal window in non OAuth login as Ftp, Sftp. Other uses a blank window in browser, so user known he is exiting from sitebuilder and useing an external storage service.

jotakar committed:
Update UnifileService.js

jotakar committed:
ModalLogin: create dialog window

Creates the layers for the modal login window
it has a nested structure: a big main layer (mBase) that contains one smaller child: mBody, this one contains three children : title (for ttal an key for close), content (that contains an iframe wit login web page) and footer that conatins close button.
Aspect is controlled by CSS.

jotakar committed:
Moved to its folder

Mistake uploading

jotakar committed:
Style for modal login 

Added style for modal login window.

jotakar committed:
Name style change

For avoid clearfy the styles change to logModal

lexoyo commented:
First it I find it a good idea to have a popin in these cases, so thank you and welcome in the team! :)

I see why you chose to create the popin with pure js, but I'm afraid it needs to be jsx code for consistency. It would be great if you could refactor it as a component? 

And to keep it in the same spirit you can expose a `.closed` attribute to your component to stay compatible with the [`startPollingAuthWin` method](https://github.com/silexlabs/CloudExplorer2/blob/master/src/js/UnifileService.js#L228) 

I believe you should use the [ModalDialog component](https://github.com/silexlabs/CloudExplorer2/blob/master/src/js/ModalDialog.jsx) and encapsulate it somehow. Let me know if you need ideas on how to do that? Maybe the great architect @clemos could give us ideas for that too?

jotakar committed:
bad background set

Modify background  color set

jotakar commented:
Hi again :)
Your suggestion is modify ModalDialog ... I need ideas about it, and my first step is to find a good JSX- React manual. :)

jotakar committed:
Update style.scss

clemos commented:
It's a tricky one :S

jotakar commented:
Well, I cannot do it with modaldialog. 
I'm tring to open a dialog window using ModalDialog, but not possible.
My problem:
I can create an instance of ModalDilog when I need open the window login, so I could create the win object to the rest of the auth process. Javascript is fine and works OK, but ModalDialog ... I jsut need that open the modal when  I need, and read from it thw windo object for the auth process.
How could I do?
I ee that error windows open inmediately, so it's posible but how?
Any help, please.

jotakar committed:
Use asynchronous callbak

For get login change state of a  ModalDialog instance, with a callback funtion that will complete the preparation of auth process.

jotakar committed:
Not use ModalLogin

Abandonde the pure JS version so this file is not necessary

jotakar commented:
I think I found the solution.
React: it's asyncrhonous yes or yes, imposible to find a way to use as sinchronous (the synchronous solution would be easiest in this case). 
Then change our minds to asyncronous mode, and the solution came!, at least I think so.

jotakar committed:
sintax error

Fixing errors

jotakar committed:
ModalDialog for login

Added section and layers for a login window, that includes am iframe with the login web page. Includes two function
One, setwin, it inits a state value (winOb) with the url of the window in the iframe.
Other getwin, for reading the actual value of the Url of the winfows in the iframe and close if the page is callback page, because in this moment the login form has been sent.

jotakar committed:
UnifileService: Bad import

Bad import, must import ModalDialog

jotakar commented:
What about this request? is it fine to be commited?

clemos commented on src/js/UnifileService.js:
```diff
@@ -185,37 +186,57 @@ export default class UnifileService {
+                var m = ModalDialog.getInstance();
+                //Change state and callback function for login in win popup
+                //Cannot have an synchronous render so need to use callback.
+                m.setState({mode:'login'},
```
I'm not sure I fully understand, but I think you need to split this in two here :
* just setState({}) here
* move the rest of the logic to the `render()` function (`if(this.state.mode === 'login')`)

jotakar replied:
Thanks for your review.
My idea was that setState({mode:'login') throws a render of modaldialog, that check the mode is login and shows the window popup. Indeed in modaldialog the login of check state is working now.
I do not know how to use just setState({}), how tell modaldialgo the state is login?
So I cannot modify this code.

clemos replied:
Ok I guess it's too much of a rework.
I really dislike that you need to `ModalDialog.getInstance()` here (I think `getInstance()` is really an anti-pattern here...), but I have no clue as to how to avoid this :S

jotakar committed:
Unifile: changes for consistencys and simplify code

win.addEventListener(unload) change to win.onunload(), is the way rest of the file uses
prepLogin(), is just for avoid repeat code, but for clearfy: it has been bint to unifile and converted to promise.

jotakar commented on src/js/UnifileService.js:
```diff
@@ -185,37 +186,57 @@ export default class UnifileService {
+			  if (req.responseText) {
+          win.location = req.responseText;
+          win.addEventListener('unload', () => {
+                win.onunload = null;
```
Ok, changed.

jotakar commented on src/js/UnifileService.js:
```diff
@@ -185,37 +186,57 @@ export default class UnifileService {
+           if (service.isOAuth){
+                const win = window.open();
+                const req = new XMLHttpRequest();
+                prepLogin(win,req,resolve, reject, UnifileService, serviceName, this);
```
Change: promise bint and retrun promise.

lexoyo commented:
Hi any news?

jotakar commented:
There isn't any change to do, and I see it's approved, so it's ready to be commited.
Bye

jotakar committed:
Merge branch 'master' into jotakar-ModalLog

jotakar commented:
with last changes the modal doesn't work fine. I need to send some modifications.

lexoyo commented:
Ok, I'll test it when you do and will merge it then

lexoyo commented:
Sorry about that @jotakar
